apps/api-extractor/src/extractor/IExtractorConfig.ts

```diff
@@ -142,6 +142,57 @@ export interface IExtractorApiJsonFileConfig {
   outputFolder?: string;
 }
 
+/**
+ * Configures how the package typings (*.d.ts) will be generated.
+ * @remarks
+ * API Extractor can generate a single unified *.d.ts file that contains all
+ * the exported typings for the package entry point.  It can also remove
+ * \@alpha \@beta \@internal definitions depending on the release type.
+ *
+ * @beta
+ */
+export interface IExtractorPackageTypingsConfig {
+  /**
+   * Whether to generate package typings.  The default is false.
+   */
+  enabled: boolean;
+
+  /**
+   * Specifies where the *.d.ts files should be written.
+   *
+   * The default value is "./dist"
+   */
+  outputFolder?: string;
+
+  /**
+   * Specifies the output filename for an internal release.
+   *
+   * @remarks
+   * The default value is "index-internal.d.ts".  This file will contain all definitions that
+   * are reachable from the entry point.
+   */
+  internalFilename?: string;
+
+  /**
+   * Specifies the output filename for a preview release.
+   *
+   * @remarks
+   * The default value is "index-preview.d.ts".  This file will contain all definitions that
+   * are reachable from the entry point, except definitions marked as \@alpha or \@internal.
+   */
+  previewFilename?: string;
+
+  /**
+   * Specifies the output filename for a public release.
+   *
+   * @remarks
+   * The default value is "index-public.d.ts".  This file will contain all definitions that
+   * are reachable from the entry point, except definitions marked as \@beta, \@alpha,
+   * or \@internal.
+   */
+  publicFilename?: string;
```

Comment:
My first guess is that these paths are inputs. Make sure it's clear from the names that they're outputs.